### ISSUE TITLE
require-default-props: Configure for rule is invalid

### ISSUE THREAD

**Issue by troncoso** [jsx-eslint/eslint-plugin-react#3290]
Hi there. I am trying to use `react/require-default-props` to ensure that non-required fields have defaults set as argument defaults, like this:

```typescript
interface ComponentProps {
    prop1: number;
    prop2: string;
    prop3?: string;
}
function Component({ prop1, prop2, prop3 = 'default-value' }) {}
```

The [documentation rule options](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-default-props.md#rule-options) state that I can do that like this:

```javascript
{
    rules: {
        'react/require-default-props': [ 'error', { functions: 'defaultArguments' } ]
    }
}
```

However, eslint says this is invalid:

```
Error: .eslintrc.js:
        Configuration for rule "react/require-default-props" is invalid:
        Value {"functions":"defaultArguments"} should NOT have additional properties.
```

Is the documentation out of date or am I just interpreting how to use it incorrectly? Thanks!

**Comment by ljharb** [comment]
This was just released in v7.30.0. What version are you using?

Always check the docs on the tag matching the version you're using, not on master - everywhere on github.

**Comment by troncoso** [comment]
Yep, I was a bit behind on my version. Sorry about that. I just set the project up this week, so I didn't have it in mind that I would end up needing to update for this feature that just released 20 hours ago at the time of this comment. Cheers!
